Demo/Demo/ViewModels/PayPalViewModel.swift

```diff
@@ -0,0 +1,97 @@
+import UIKit
+import PayPalNativeCheckout
+import PayPalCheckout
+import PaymentsCore
+import Card
+
+class PayPalViewModel: ObservableObject, PayPalDelegate {
+
+    enum State {
+        case initial
+        case loading(content: String)
+        case mainContent(title: String, content: String, flowComplete: Bool)
+    }
+
+    @Published private(set) var state = State.initial
+
+    private var accessToken = ""
+    private var payPalClient: PayPalClient?
+
+    func getAccessToken() {
+        Task {
+            state = .loading(content: "Getting access token")
+            guard let token = await getAccessToken() else {
+                state = .mainContent(title: "Error", content: "Failed to fetch access token", flowComplete: true)
+
+                return
+            }
+            accessToken = token
+            payPalClient = PayPalClient(config: CoreConfig(accessToken: token, environment: PaymentsCore.Environment.sandbox))
+            payPalClient?.delegate = self
+            state = .mainContent(title: "Access Token", content: accessToken, flowComplete: false)
+        }
+    }
+
+    func retry() {
+        payPalClient = nil
+        accessToken = ""
+        state = .initial
+    }
+
+    func checkoutWithOrderID() {
+        startNativeCheckout {
+            let orderID = try await self.getOrderIDWithFixedShipping()
+            await self.payPalClient?.start { createOrderAction in
+                createOrderAction.set(orderId: orderID)
+            }
+        }
+    }
+
+    private func getOrderIDWithFixedShipping() async throws -> String {
+        let order = try await DemoMerchantAPI.sharedService.createOrder(
+            orderParams: OrderRequestHelpers.requestWithFixedShipping
+        )
+        return order.id
+    }
+
+    private func startNativeCheckout(withAction action: @escaping () async throws -> Void) {
```

Comment:
Would `_ completion` be a better name for this param? `withAction action` took me a few to understand and look where it was used to know what was going on.